pkg/load/configAgent.go

```diff
@@ -0,0 +1,151 @@
+package load
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/coalescer"
+	"github.com/openshift/ci-tools/pkg/config"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/interrupts"
+)
+
+type ConfigAgent interface {
+	GetConfig(config.Info) (api.ReleaseBuildConfiguration, error)
+	GetGeneration() int
+}
+
+type filenameToConfig map[string]api.ReleaseBuildConfiguration
+
+type agent struct {
+	lock       sync.Locker
+	configs    filenameToConfig
+	configPath string
+	cycle      time.Duration
+	generation int
+}
+
+func NewConfigAgent(configPath string, cycle time.Duration) (ConfigAgent, error) {
+	a := &agent{configPath: configPath, cycle: cycle}
+	// mutex to prevent data race on o.configs
+	a.lock = &sync.Mutex{}
+	configCoalescer := coalescer.NewCoalescer(a.loadFilenameToConfig)
+	err := configCoalescer.Run()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to load configs: %v", err)
+	}
+
+	// periodic reload
+	if a.cycle != 0 {
```

Comment:
What's the use in disabling this?

Reply:
Not sure if there is much reason to do so. Should I just add a check in `validateOptions` for the `configresolver` that errors out if the cycle is 0 instead?

Reply:
Sure, and default to something sensible so we can always run it